inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -195,6 +194,20 @@ def __init__(self,
                  params: Dict) -> None:
         super().__init__(dispatcher_broker, repo, params)
 
+    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
+        # get the GUID from the system using FW tool
+        extracted_guids = extract_guids(
+            self._fw_tool, ["System Firmware type", "system-firmware type"])
+        if manifest_guid:
+            if manifest_guid not in extracted_guids:
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

Instead of manually checking and raising an error for `manifest_guid` not in `extracted_guids`, consider using a more Pythonic approach with a try-except block around list.index().